### PR TITLE
Initial doc for state next

### PR DESCRIPTION
**Pull request by moritzraho**
## Description

Developer preview documentation for State Next

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by shazron — APPROVED** [review]

**Review by pru55e11 — APPROVED** [review]

File: src/pages/guides/application_state.md [hunk 1]
```diff
@@ -33,29 +32,116 @@ NAME=Joe
...
+npm install @adobe/aio-lib-state@next
+```
+
+```js
+  const stateLib = require('@adobe/aio-lib-state')
+
+  // init with implicit I/O Runtime credentials, default region is 'amer'.
+  const state = await stateLib.init()
+  // set an explicit region
+  const state2 = await stateLib.init({ region: 'emea' })
```
Comment (purplecabbage) [inline src/pages/guides/application_state.md:81]: Do we define regions somewhere? What regions are available? Why would I choose one over another?

File: src/pages/guides/application_state.md [hunk 2]
```diff
@@ -33,29 +32,116 @@ NAME=Joe
...
+
+Explore the [full API](https://github.com/adobe/aio-lib-state/blob/main/doc/api.md)
+
+***CLI usage, from your local machine***: *coming soon!*
+
+### Limits & validation (preview)
+
+Limits are enforced and can't be changed on a per-user basis.
+
+- State is only available to Runtime Namespaces that follow the AppBuilder format: `amsorg-project(-workspace)?`.
```
Comment (purplecabbage) [inline src/pages/guides/application_state.md:110]: What is amsorg?  I don't work at Adobe ... please help me?

File: src/pages/guides/application_state.md [hunk 3]
```diff
@@ -33,29 +32,116 @@ NAME=Joe
...
 
-## State and files persistence at runtime
+## Persistence at runtime
 
-As part of App Builder, you will have out-of-the-box access to file storage and to a key-value store. These are particularly useful when you want to persist data dynamically in the individual action invocations.
+As part of App Builder, you will have out-of-the-box access to *Files* and *State*, our two storage services meant for persisting data dynamically from your Runtime actions.
 
-To provide zero-config state and file caching for App Builder, we have created the [Adobe I/O Files library](https://github.com/adobe/aio-lib-files) and [Adobe I/O State library](https://github.com/adobe/aio-lib-state). The Adobe I/O State library is an npm module that provides a JavaScript abstraction on top of distributed/cloud DBs with a simple key-value store state persistence API; whereas the Adobe I/O Files library provides a JavaScript abstraction on top of cloud blob storages with a simple file-system like persistence API.
+No pre-configuration is required, just install the libraries and use them in your project. We will be transparently using your AppBuilder credentials to authorize and entitle your requests.
```
Comment (purplecabbage) [inline src/pages/guides/application_state.md:49]: App Builder

File: src/pages/guides/application_state.md (hunk 2 above)
Comment (purplecabbage) [reply src/pages/guides/application_state.md:110]: App Builder

File: src/pages/guides/application_state.md [hunk 4]
```diff
@@ -33,29 +32,116 @@ NAME=Joe
...
+- State is only available to Runtime Namespaces that follow the AppBuilder format: `amsorg-project(-workspace)?`.
+- Max state value size: `1MB`.
+- Max state key size: `1024 bytes`.
+- Max-supported TTL is `365 days`.
+- Values format: any `string|binary`.
+- Keys format: `string` only `alphanumeric` with `-`,`_`,`.`.
+
+### Quotas (preview)
+
+Quotas are limits that depend on the organization's entitlements. Every organization with AppBuilder access is entitled to at least 1 State quota.
```
Comment (purplecabbage) [inline src/pages/guides/application_state.md:119]: App Builder

File: src/pages/guides/application_state.md (hunk 1 above)
Comment (moritzraho) [reply src/pages/guides/application_state.md:81]: is the `how is my data stored` intro enough for now? 
https://adobedocs.github.io/app-builder/guides/application_state/#state

Comment (moritzraho) [reply src/pages/guides/application_state.md:81]: will add something more on the bullet point

**Pull request merged**